python/lsst/cp/verify/verifyPtc.py

```diff
@@ -0,0 +1,211 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import numpy as np
+import lsst.pex.config as pexConfig
+
+import lsst.pipe.base.connectionTypes as cT
+from .verifyCalib import CpVerifyCalibConfig, CpVerifyCalibTask, CpVerifyCalibConnections
+
+__all__ = ['CpVerifyPtcConnections', 'CpVerifyPtcConfig', 'CpVerifyPtcTask']
+
+
+class CpVerifyPtcConnections(CpVerifyCalibConnections,
+                             dimensions={"instrument", "detector"},
+                             defaultTemplates={}):
+    inputCalib = cT.Input(
+        name="calib",
+        doc="Input calib to calculate statistics for.",
+        storageClass="PhotonTransferCurveDataset",
+        dimensions=["instrument", "detector"],
+        isCalibration=True
+    )
+
+
+class CpVerifyPtcConfig(CpVerifyCalibConfig,
+                        pipelineConnections=CpVerifyPtcConnections):
+    """Inherits from base CpVerifyCalibConfig."""
+
+    def setDefaults(self):
+        super().setDefaults()
+
+    gainThreshold = pexConfig.Field(
+        dtype=float,
+        doc="Maximum percentage difference between PTC gain and nominal amplifier gain.",
+        default=5.0,
+    )
+
+    noiseThreshold = pexConfig.Field(
+        dtype=float,
+        doc="Maximum percentage difference between PTC readout noise and nominal "
+            "amplifier readout noise.",
+        default=5.0,
+    )
+
+    turnoffThreshold = pexConfig.Field(
+        dtype=float,
+        doc="Minimun full well requirement (in electrons). To be compared with the "
+            "reported PTC turnoff per amplifier.",
+        default=90000,
+    )
+
+    a00MinITL = pexConfig.Field(
+        dtype=float,
+        doc="Minimum a00 (c.f., Astier+19) for ITL CCDs.",
+        default=-4.56e-6,
+    )
+
+    a00MaxITL = pexConfig.Field(
+        dtype=float,
+        doc="Maximum a00 (c.f., Astier+19) for ITL CCDs.",
+        default=6.91e-7,
+    )
+
+    a00MinE2V = pexConfig.Field(
+        dtype=float,
+        doc="Minimum a00 (c.f., Astier+19) for E2V CCDs.",
+        default=-3.52e-6,
+    )
+
+    a00MaxE2V = pexConfig.Field(
+        dtype=float,
+        doc="Maximum a00 (c.f., Astier+19) for E2V CCDs.",
+        default=-2.61e-6,
+    )
+
+
+class CpVerifyPtcTask(CpVerifyCalibTask):
+    """PTC verification sub-class, implementing the verify method.
+    """
+    ConfigClass = CpVerifyPtcConfig
+    _DefaultName = 'cpVerifyPtc'
+
+    def detectorStatistics(self, inputCalib):
+        """Calculate detector level statistics from the calibration.
+
+        Parameters
+        ----------
+        inputCalib : `lsst.ip.isr.IsrCalib`
+            The calibration to verify.
+
+        Returns
+        -------
+        outputStatistics : `dict` [`str`, scalar]
+            A dictionary of the statistics measured and their values.
+        """
+        return {}
```

Comment:
Can this be used to store the values that are being tested?  I know this duplicates information from the PTC dataset, but it will make it easier to migrate that information into a database in the future.

Reply:
Yes. Is there an example of this in other "verify" tests? Should they be called something like "PTC_GAIN", "PTC_NOISE", etc.

Reply:
I'm thinking about putting these in a new ``amplifierStatistics`` method since the relevant quantities are per amp (gain, turnoff, noise, a00).